internal/driver/nodeserver.go

```diff
@@ -276,6 +278,25 @@ func (ns *LinodeNodeServer) NodeUnstageVolume(ctx context.Context, req *csi.Node
 func (ns *LinodeNodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolumeRequest) (*csi.NodeExpandVolumeResponse, error) {
 	klog.V(4).Infof("NodeExpandVolume called with req: %#v", req)
 
+	// Validate req (NodeExpandVolumeRequest)
+	if err := validateNodeExpandVolumeRequest(req); err != nil {
+		return nil, err
+	}
+
+	// Check linode to see if a give volume exists by volume ID
+	// Make call to linode api using the lindoe api client
+	LinodeVolumeKey, err := common.ParseLinodeVolumeKey(req.GetVolumeId())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Error parsing linode volume key: %v", err)
+	}
+	jsonFilter, err := json.Marshal(map[string]string{"label": LinodeVolumeKey.Label})
+	if err != nil {
+		return nil, errInternal("marshal json filter: %v", err)
+	}
+	if _, err = ns.CloudProvider.ListVolumes(ctx, linodego.NewListOptions(0, string(jsonFilter))); err != nil {
```

Comment:
Not sure how many times this api call will be made (thinking just once or twice when we run the tests), if its very few times, then we are ok. If this call will be made periodically every few mins, then we need to discuss if we should introduce additional calls as we we don't do anything with such requests and simply return. I am hoping its just for tests, so this should be fine. Let me know if there are any other cases when this will be triggered.

Reply:
The Api call isn't made very often. In this case, the test never really reaches it either.